File: src/test/java/it/marketplace/microservices/database/repository/PaymentInstallmentsRepositoryTest.java
package it.marketplace.microservices.database.repository;

import it.marketplace.microservices.common.enums.StatusOrderEnum;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import java.time.LocalDateTime;


import static org.junit.jupiter.api.Assertions.*;

@DataJpaTest
class PaymentInstallmentsRepositoryTest {

    @Autowired
    private PaymentInstallmentsRepository repository;

    @Test
    void shouldFindByOrderCode_ArrangeActAssert() {
        // Arrange
        PaymentInstallmentsEntity entity = new PaymentInstallmentsEntity();
        entity.setOrderCode("ORD1");
        entity.setStatus(StatusOrderEnum.CREATED);
        entity.setDebit(0.59);
        entity.setReference("ORDER");
        entity.setTmsUpdate(LocalDateTime.now());
        repository.save(entity);
        // Act
        List<PaymentInstallmentsEntity> found = repository.findByOrderCode("ORD1");
        // Assert
        assertFalse(found.isEmpty());
        assertEquals("ORD1", found.get(0).getOrderCode());
    }

    @Test
    void shouldFindByOrderCodeAndStatus_ArrangeActAssert() {
        // Arrange
        PaymentInstallmentsEntity entity = new PaymentInstallmentsEntity();
        entity.setOrderCode("ORD2");
        entity.setStatus(StatusOrderEnum.PAID);
        entity.setDebit(0.59);
        entity.setReference("ORDER");
        entity.setTmsUpdate(LocalDateTime.now());
        repository.save(entity);
        // Act
        List<PaymentInstallmentsEntity> found = repository.findByOrderCodeAndStatus("ORD2", StatusOrderEnum.PAID);
        // Assert
        assertFalse(found.isEmpty());
        assertEquals(StatusOrderEnum.PAID, found.get(0).getStatus());
    }
}

File: adapter/output/database/src/test/kotlin/it.marketplace/repository/PaymentInstallmentsRepositoryTest.kt
package it.marketplace.database.repository

import it.marketplace.common.enums.StatusOrderEnum
import it.marketplace.database.entity.PaymentInstallmentsEntity
import it.marketplace.utils.BaseTest
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import java.time.LocalDateTime
import kotlin.test.assertEquals
import kotlin.test.assertFalse

@DataJpaTest
open class PaymentInstallmentsRepositoryTest : BaseTest() {

    @Autowired
    private val repository: PaymentInstallmentsRepository = mock()

    @Test
    fun shouldFindByOrderCode_ArrangeActAssert() {
        // Arrange
        val now = LocalDateTime.now()
        val entity = mockPaymentInstallmentsEntity()
        entity.orderCode = "ORD1"
        entity.status = StatusOrderEnum.CREATED
        entity.debit = 0.59
        entity.reference = "ORDER"
        entity.tmsUpdate = LocalDateTime.now()
        repository.save(entity)
        // Act
        val found: List<PaymentInstallmentsEntity?> = repository.findByOrderCode("ORD1")
        // Assert
        assertFalse(found.isEmpty())
        assertEquals("ORD1", found[0]?.orderCode)
    }

    @Test
    fun shouldFindByOrderCodeAndStatus_ArrangeActAssert() {
        // Arrange
        val now = LocalDateTime.now()
        val entity = mockPaymentInstallmentsEntity()
        entity.orderCode = "ORD2"
        entity.status = StatusOrderEnum.PAID
        entity.debit = 0.59
        entity.reference = "ORDER"
        entity.tmsUpdate = LocalDateTime.now()
        repository.save(entity)
        // Act
        val found: List<PaymentInstallmentsEntity?> =
            repository.findByOrderCodeAndStatus("ORD2", StatusOrderEnum.PAID)
        // Assert
        assertFalse(found.isEmpty())
        assertEquals(StatusOrderEnum.PAID, found[0]?.status)
    }
}

